### PR TITLE
Migrate tests from "web/oauth_test.go" to use testify

### LINKED ISSUE #12793: Migrate tests from "web/oauth_test.go" to use testify
Mattermost is migrating its server tests from using the `t.Fatal` method calls on their checks to use the [testify toolkit](https://github.com/stretchr/testify). The goal of this migration is to improve the readability of the tests, making them more concise and meaningful. This Help Wanted issue is to migrate the tests in the `web/oauth_test.go` file.

The expected way to migrate this is to go to the `web/oauth_test.go` file and modify the calls to `t.Fatal` and the condition checks that lead to them, replacing them with calls to the [the `require` package](https://godoc.org/github.com/stretchr/testify/require] if a failure should stop the execution of the test, or [the `assert` package|https://godoc.org/github.com/stretchr/testify/assert) if it shouldn't.

Example: mattermost/mattermost-server#12039

If you have questions about the ticket or you need any help, feel free to contact `miguel.delacruz` or `jesus.espino` in <https://community.mattermost.com/>

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-19462

### PR DESCRIPTION
**Pull request by bensooraj**
Summary

Migrate tests from "store/storetest/reaction_store.go" to use testify

Ticket Link

Fixes #12793

### CODE REVIEW

File: web/oauth_test.go [hunk 1]
```diff
@@ -617,35 +563,27 @@ func GenerateTestAppName() string {
 func CheckNoAppError(t *testing.T, err *model.AppError) {
 	t.Helper()
 
-	if err != nil {
-		t.Fatalf("Expected no error, got %q", err.Error())
-	}
+	require.Nil(t, err, "Expected no error, got %q", err.Error())
```
Comment (hanzei) [inline web/oauth_test.go]: This is causing an NPE. Would you please just remove the function in whole and just use `require.Nil`?

Comment (bensooraj) [reply web/oauth_test.go]: Hey @hanzei, thanks for the review.

I tried to follow your suggestion, but the build has failed again. Can you help me understand what could have possibly gone wrong?

File: web/oauth_test.go [hunk 2]
```diff
@@ -65,7 +65,7 @@ func TestAuthorizeOAuthApp(t *testing.T) {
 	}
 
 	rapp, appErr := th.App.CreateOAuthApp(oapp)
-	CheckNoAppError(t, appErr)
+	require.Nil(t, appErr, "Expected no error, got %q", appErr.Error())
```
Comment (hanzei) [inline web/oauth_test.go]: This will panic. You need to just use 
```suggestion
	require.Nil(t, appErr)
```

**Comment by hanzei** [comment]
Hey @bensooraj, you still have a NPE in your code. Let me know, if my comment helps fixing it.

**Comment by bensooraj** [comment]
Hey @hanzei! Thanks for your suggestions. Finally, all checks have passed.

If it's okay with you, can you please help me understand the difference between the two? Why should the first one raise an NPE?
```go
require.Nil(t, appErr, "Expected no error, got %q", appErr.Error())
// And...
require.Nil(t, appErr)
```

File: web/oauth_test.go (hunk 1 above)
Comment (hanzei) [reply web/oauth_test.go]: I looked into but didn't found the cause either. :man_shrugging:

File: web/oauth_test.go [hunk 3]
```diff
@@ -77,28 +77,19 @@ func TestAuthorizeOAuthApp(t *testing.T) {
 
 	// Test auth code flow
 	ruri, resp := ApiClient.AuthorizeOAuthApp(authRequest)
-	CheckNoError(t, resp)
+	require.Nil(t, resp.Error)
 
-	if len(ruri) == 0 {
-		t.Fatal("redirect url should be set")
-	}
+	require.NotZero(t, len(ruri), "redirect url should be set")
```
Comment (jespino) [inline web/oauth_test.go]: Probably is more sematic here to use `require.NotEmpty` instead of `NotZero` + `len(ruri)`

File: web/oauth_test.go [hunk 4]
```diff
@@ -77,28 +77,19 @@ func TestAuthorizeOAuthApp(t *testing.T) {
...
-	} else {
-		if len(ru.Query().Get("code")) == 0 {
-			t.Fatal("authorization code not returned")
-		}
-		if ru.Query().Get("state") != authRequest.State {
-			t.Fatal("returned state doesn't match")
-		}
-	}
+	require.NotNil(t, ru, "redirect url unparseable")
+	require.NotZero(t, len(ru.Query().Get("code")), "authorization code not returned")
```
Comment (jespino) [inline web/oauth_test.go]: Same here, probably better to use `require.NotEmpty`

File: web/oauth_test.go [hunk 5]
```diff
@@ -297,101 +277,67 @@ func TestOAuthAccessToken(t *testing.T) {
...
-			t.Fatal("access token type incorrect")
-		}
-	}
-
-	if _, err := ApiClient.DoApiGet("/oauth_test", ""); err != nil {
-		t.Fatal(err)
-	}
+	rsp, resp := ApiClient.GetOAuthAccessToken(data)
+	require.Nil(t, resp.Error)
+	require.NotZero(t, len(rsp.AccessToken), "access token not returned")
```
Comment (jespino) [inline web/oauth_test.go]: `require.NotEmpty`

File: web/oauth_test.go [hunk 6]
```diff
@@ -297,101 +277,67 @@ func TestOAuthAccessToken(t *testing.T) {
...
+	require.NotZero(t, len(rsp.RefreshToken), "refresh token not returned")
```
Comment (jespino) [inline web/oauth_test.go]: `require.NotEmpty`

File: web/oauth_test.go [hunk 7]
```diff
@@ -297,101 +277,67 @@ func TestOAuthAccessToken(t *testing.T) {
...
-		if len(rsp.AccessToken) == 0 {
-			t.Fatal("access token not returned")
-		} else if len(rsp.RefreshToken) == 0 {
-			t.Fatal("refresh token not returned")
-		} else if rsp.RefreshToken == refreshToken {
-			t.Fatal("refresh token did not update")
-		}
+	rsp, resp = ApiClient.GetOAuthAccessToken(data)
+	require.Nil(t, resp.Error)
+	require.NotZero(t, len(rsp.AccessToken), "access token not returned")
```
Comment (jespino) [inline web/oauth_test.go]: `require.NotEmpty`

File: web/oauth_test.go [hunk 8]
```diff
@@ -297,101 +277,67 @@ func TestOAuthAccessToken(t *testing.T) {
...
+	require.NotZero(t, len(rsp.RefreshToken), "refresh token not returned")
```
Comment (jespino) [inline web/oauth_test.go]: `require.NotEmpty`

File: web/oauth_test.go [hunk 9]
```diff
@@ -297,101 +277,67 @@ func TestOAuthAccessToken(t *testing.T) {
...
-			t.Fatal("access token type incorrect")
-		}
-		ApiClient.SetOAuthToken(rsp.AccessToken)
-		if _, err := ApiClient.DoApiGet("/oauth_test", ""); err != nil {
-			t.Fatal(err)
-		}
+	data.Set("refresh_token", rsp.RefreshToken)
+	rsp, resp = ApiClient.GetOAuthAccessToken(data)
+	require.Nil(t, resp.Error)
+	require.NotZero(t, len(rsp.AccessToken), "access token not returned")
```
Comment (jespino) [inline web/oauth_test.go]: `require.NotEmpty`

File: web/oauth_test.go [hunk 10]
```diff
@@ -297,101 +277,67 @@ func TestOAuthAccessToken(t *testing.T) {
...
+	require.NotZero(t, len(rsp.RefreshToken), "refresh token not returned")
```
Comment (jespino) [inline web/oauth_test.go]: `require.NotEmpty`

**Comment by hanzei** [comment]
@bensooraj Do you have any questions about the feedback? Let me know how I can help

**Comment by jasonblais** [comment]
Hi @bensooraj, it's been a while since we've had activity on this PR, so I'll close this PR for now following our [inactive contributions process](https://developers.mattermost.com/contribute/getting-started/inactive-contributions/). If you'd like to work on it, let us know! We'd be happy to guide you through the feedback above :)

**Pull request closed without merging**